src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -2838,5 +2838,97 @@ private static unsafe int FromBase64_ComputeResultLength(char* inputPtr, int inp
+        /// <returns>The string representation in hex of the elements in <paramref name="inArray"/>.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="inArray"/> is <code>null</code>.</exception>
+        public static string ToHexString(byte[] inArray)
+        {
+            if (inArray == null)
+                throw new ArgumentNullException(nameof(inArray));
+
+            return ToHexString(new ReadOnlySpan<byte>(inArray));
+        }
+
+        /// <summary>
+        /// Converts a subset of an array of 8-bit unsigned integers to its equivalent string representation that is encoded with hex characters.
+        /// Parameters specify the subset as an offset in the input array and the number of elements in the array to convert.
+        /// </summary>
+        /// <param name="inArray">An array of 8-bit unsigned integers.</param>
+        /// <param name="offset">An offset in <paramref name="inArray"/>.</param>
+        /// <param name="length">The number of elements of <paramref name="inArray"/> to convert.</param>
+        /// <returns>The string representation in hex of <paramref name="length"/> elements of <paramref name="inArray"/>, starting at position <paramref name="offset"/>.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="inArray"/> is <code>null</code>.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="offset"/> or <paramref name="length"/> is negative.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="offset"/> plus <paramref name="length"/> is greater than the length of <paramref name="inArray"/>.</exception>
+        public static string ToHexString(byte[] inArray, int offset, int length)
+        {
+            if (inArray == null)
+                throw new ArgumentNullException(nameof(inArray));
+            if (length < 0)
+                throw new ArgumentOutOfRangeException(nameof(length), SR.ArgumentOutOfRange_Index);
+            if (offset < 0)
+                throw new ArgumentOutOfRangeException(nameof(offset), SR.ArgumentOutOfRange_GenericPositive);
+            if (offset > (inArray.Length - length))
+                throw new ArgumentOutOfRangeException(nameof(offset), SR.ArgumentOutOfRange_OffsetLength);
+
+            return ToHexString(new ReadOnlySpan<byte>(inArray, offset, length));
+        }
+
+        /// <summary>
+        /// Converts a span of 8-bit unsigned integers to its equivalent string representation that is encoded with hex characters.
+        /// </summary>
+        /// <param name="bytes">A span of 8-bit unsigned integers.</param>
+        /// <returns>The string representation in hex of the elements in <paramref name="bytes"/>.</returns>
+        public static string ToHexString(ReadOnlySpan<byte> bytes)
+        {
+            if (bytes.Length == 0)
+                return string.Empty;
+            if ((long)bytes.Length * 2 > int.MaxValue)
```

Comment:
I think you could just make this:

```suggestion
            if (bytes.Length > int.MaxValue / 2)
```

And Roslyn should constantize that to 1073741823.

Reply:
Good catch. I'll address that shortly...

See diff here: [https://sharplab.io](https://sharplab.io/#v2:C4LghgzgtgPgAgJgIwFgBQcAMACOSB0ASgK4B2wAllAKb4DCA9lAA4UA21ATgMpcBuFAMbUIAbnTo4AZlwJsdbAG902Vbhl4AbLgAs8gBbVBAa27MwpADLVSAc2D6kACkLUwAEwDypNgE8zFgA8AEa+wNQAfNih4RAAlCpqymhqqdgUAGbYTk5sDHZxMSL41nYO2ABU2HJRFOT4ALJgAB4AamBsxNQJKWl9ACr6nAwA7k5x4r2qAL6JqnPquEjacHp0hiYBVjb2+ggubl4+/uakIWGR0RfxC8l9qpnZRRAlO+W19U1tHV3YAPTVHr3NKDYZjCYLWZTbALADaDWoDgY7gAkiw2E4EUjUejPMxKPkXgA5Bgonx1Oq2OIAXQWzE4FD4YHCSxWelBo3GSih0yAA=)